graphql-language/src/ast.rs

```diff
@@ -7,22 +7,23 @@ use source::Source;
 
 /// Contains some character offsets that identify where the source of the AST
 /// is from.
+#[derive(Clone, Copy)]
 pub struct Location<'a> {
   pub start: usize,
   pub end: usize,
-  pub source: Option<&'a Source<'a>>
+  pub source: Source<'a>
```

Comment:
That is a large breach of concerns though, especially across module bounds. Now we depend on `Source` to only ever be 2 pointers, and if we ever need to switch the `&str`s to `String` like you recommended, there would be problems.